File: 1.coding/binary trees/levelorder.cpp
int height(TreeNode* root)
{
    if(root == NULL)return 0;
    return 1+ max(height(root->left), height(root->right));
}

vector<int> curLevel(TreeNode* root, int h, vector<int>&a)
{
    if(root == NULL)
        return a;
    if(h == 1)
    {
        a.push_back(root->val);
        //return a;
    }
    else if(h > 1)
    {
        curLevel(root->left, h-1, a);
        curLevel(root->right, h-1, a);
    }
    return a;
}
vector<vector<int>> levelOrder(TreeNode* root) 
{
    vector<vector<int>> ans;
    int h = height(root);
    for(int i=1; i<=h; i++)
    {
        vector<int>a;
        curLevel(root, i, a);
        ans.push_back(a);
    }
    return ans;
    
}

File: 1.coding/binary trees/levelOrder.java
import java.util.*;
import java.lang.Math;
class Node 
{
	int data;
	Node left, right;
	public Node(int item)
	{
		data = item;
		left = right = null;
	}
}

class BinaryTree 
{
	// Root of the Binary Tree
	Node root;

	public BinaryTree() { root = null; }

	void printLevelOrder()
	{
		int h = height(root);
		int i;
		for (i = 1; i <= h; i++)
			printCurrentLevel(root, i);
	}


	int height(Node root)
	{
		if (root == null)
			return 0;
		else 
        {
            return 1 + Math.max(height(root.left), height(root.right));
		}
	}

	/* Print nodes at the current level */
	void printCurrentLevel(Node root, int level)
	{
		if (root == null)
			return;
		if (level == 1)
			System.out.print(root.data + " ");
		else if (level > 1) {
			printCurrentLevel(root.left, level - 1);
			printCurrentLevel(root.right, level - 1);
		}
	}

	public static void main(String args[])
	{
		BinaryTree tree = new BinaryTree();
		tree.root = new Node(3);
		tree.root.left = new Node(9);
		tree.root.right = new Node(20);
		tree.root.right.left = new Node(15);
		tree.root.right.right = new Node(7);

        System.out.println("Level order traversal of binary tree is ");
        tree.printLevelOrder();
	}
}
